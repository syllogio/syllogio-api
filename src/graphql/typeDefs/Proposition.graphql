"""
Describes a proposition.
"""
type Proposition {
  """
  Unique identifier that represents this proposition.
  """
  id: Int!
  """
  The text containing the proposition itself.
  """
  text: String!
  """
  Unique identifier of other propositions which are supported by this
  proposition.
  """
  supports: [Int]!
  """
  Unique identifier of other propositions by which this proposition is
  supported.
  """
  supportedBy: [Int]!
}
