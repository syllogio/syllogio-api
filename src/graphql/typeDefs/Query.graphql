"""
Defines entry points for all available queries.
"""
type Query {
  """
  Get a proposition by its identifier.
  """
  proposition(id: Int!): Proposition
}
