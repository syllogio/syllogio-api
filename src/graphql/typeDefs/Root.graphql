# import Proposition from 'Proposition.graphql'

"""
Defines entry points for all available queries.
"""
type Query {
  """
  Get a proposition by its identifier.
  """
  proposition(id: Int!): Proposition
}

type Mutation {
  """
  Create a proposition.
  """
  createProposition(
    text: String!
    supports: [Int!]
    supportedBy: [Int!]
  ): Proposition
}
