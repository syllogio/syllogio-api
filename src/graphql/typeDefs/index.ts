import { importSchema } from 'graphql-import';
import { join } from 'path';

const typeDefs = importSchema(join(__dirname, './Root.graphql'));

export default typeDefs;
