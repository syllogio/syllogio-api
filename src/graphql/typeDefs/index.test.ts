import typeDefs from './index';

describe('typeDefs', () => {
  it('exports the GraphQL schema', () => {
    expect.assertions(1);
    expect(typeDefs).toMatchSnapshot();
  });
});
