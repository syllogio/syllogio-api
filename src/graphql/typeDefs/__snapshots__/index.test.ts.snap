// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeDefs exports the GraphQL schema 1`] = `
"\\"\\"\\"Defines entry points for all available queries.\\"\\"\\"
type Query {
  \\"\\"\\"Get a proposition by its identifier.\\"\\"\\"
  proposition(id: Int!): Proposition
}

type Mutation {
  \\"\\"\\"Create a proposition.\\"\\"\\"
  createProposition(text: String!, supports: [Int!], supportedBy: [Int!]): Proposition
}

\\"\\"\\"Describes a proposition.\\"\\"\\"
type Proposition {
  \\"\\"\\"Unique identifier that represents this proposition.\\"\\"\\"
  id: Int!
  \\"\\"\\"The text containing the proposition itself.\\"\\"\\"
  text: String!
  \\"\\"\\"
  Unique identifier of other propositions which are supported by this
  proposition.
  \\"\\"\\"
  supports: [Int]!
  \\"\\"\\"
  Unique identifier of other propositions by which this proposition is
  supported.
  \\"\\"\\"
  supportedBy: [Int]!
}
"
`;
